lib/jsverify.d.ts

```diff
@@ -114,6 +114,7 @@ declare namespace JSVerify {
   function oneof<A, B, C, D, E>(gs: [Arbitrary<A>, Arbitrary<B>, Arbitrary<C>, Arbitrary<D>, Arbitrary<E>]): Arbitrary<A | B | C | D | E>;
   function oneof<T>(gs: Arbitrary<T>[]): Arbitrary<T>;
   function record<T>(arbs: { [P in keyof T]: Arbitrary<T[P]> }): Arbitrary<T>;
+  function letrec<A>(tie: (f: (key: string) => Arbitrary<A>) => { [key: string]: Arbitrary<A> }): { [key: string]: Arbitrary<A> }
```

Comment:
doesn't typescript allow narrowing down `string` to only keys used in the record? I haven't followed recent developments, but recall something like that? (e.g. as in `record` above)?

Reply:
Very good remark, thanks. I just pushed a new commit to fix this.

Reply:
You could simplify it even further, if you say that the `K extends string`:

```suggestion
 function letrec<K extends string, V>(tie: (mapping: (key: K) => Arbitrary<V>) => Record<K, Arbitrary<V>>): Record<K, Arbitrary<V>>;
```